app/services/staff_action_logger.rb

```diff
@@ -803,6 +803,40 @@ def log_destroy_email(user)
     )
   end
 
+  def log_watched_words_creation(watched_word)
+    raise Discourse::InvalidParameters.new(:watched_word) unless watched_word
+
+    details = if watched_word.replacement.present?
+      "#{watched_word.word} → #{watched_word.replacement}"
+    else
+      watched_word.word
+    end
+
+    UserHistory.create!(
+      action: UserHistory.actions[:add_watched_word],
+      acting_user_id: @admin.id,
+      details: details,
+      context: WatchedWord.actions[watched_word.action]
+    )
+  end
+
+  def log_watched_words_deletion(watched_word)
+    raise Discourse::InvalidParameters.new(:watched_word) unless watched_word
+
+    details = if watched_word.replacement.present?
+      "#{watched_word.word} → #{watched_word.replacement}"
+    else
+      watched_word.word
+    end
```

Comment:
I think we should avoid repeating this code twice and refactor it into the `WatchWord` model.